cli/popper/commands/cmd_stages.py

```diff
@@ -0,0 +1,47 @@
+import click
+import os
+import popper.utils as pu
+
+from popper.cli import pass_context
+from popper.exceptions import BadArgumentUsage
+
+@click.command('stages', short_help='Comma-separated list of stages to a pipeline')
+@click.argument('pipeline', required=True)
+@click.option(
+    '--set',
+    help="Redefine the stages of a pipeline",
+    required=False
+)
+@pass_context
+def cli(ctx, pipeline, set):
+    """Manipulates the stages of a pipeline.The stages YAML entry specifies
+    the sequence of stages that are executed by the popper check command.When
+    a new pipeline is created, the popper init generates scaffold scripts for
+    setup.sh,run.sh,post-run.sh,validate.sh,teardown.sh by default.(See popper
+    init --help for more)
+
+    Example:
+
+    popper stages #Shows the stages of a pipeline
+
+    popper stages --set=one,two,three #Redefine the stages of a pipeline
+    """
+    config = pu.read_config()
+
+    if not set:
+        pu.print_yaml(str(config['pipelines'][pipeline]['stages']))
+
+    if set:
+        for s in stages.split(','):
+            s_filename = os.path.join(pipeline_path, s)
+            if not os.path.isfile(s_filename) and not os.path.isfile(s_filename+'.sh'):
+                pu.fail("Unable to find script for stage " + s +
+                ". You might need to provide values for the --set flag.")
+
+        if 'teardown' in stages and stages.split(',')[-1] != 'teardown' :
+            raise BadArgumentUsage('--stages = Teardown should be the last stage.'
+        + ' Consider renaming it or putting it at the end.')
+
+        config['pipelines'][pipeline]['stages'] = stages.split(',')
```

Comment:
It would be good to take this opportunity and refactor code so we avoid duplicating functionality. We have code that achieves the same thing also [here](https://github.com/systemslab/popper/blob/master/cli/popper/commands/cmd_init.py#L86) and [here](https://github.com/systemslab/popper/blob/master/cli/popper/commands/cmd_init.py#L121). It would be better if we refactor it and put it in `utils.py` so that we don't have duplicated code.

Reply:
@ivotron It's the same as the code in the initialize_existing_pipeline function, so I can make a new function check_stages in utils.py which does the same.However, incase of initialize_new_pipeline function, it writes new bash scripts,How do we refactor that?

Reply:
this can be a `get_stages(stages)` function that takes a comma-separated list of stage names and returns a list with the name of the stages. This list is used differently on each of these 3 different places

Reply:
@ivotron I have made some changes. Is this what you meant to do?